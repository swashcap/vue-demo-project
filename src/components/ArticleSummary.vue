<template>
  <article>
    <router-link :to="'/article/' + id">
      <h1>{{title}}</h1>
      <p>{{summary}}</p>
    </router-link>
  </article>
</template>

<script>
export default {
  name: 'app-article',
  props: ['id', 'summary', 'title']
}
</script>

<style scoped>
article > a {
  display: block;
  padding: 1em;
  text-decoration: none;
}
h1 {
  display: inline-block;
  font-size: 100%;
  font-weight: 600;
  margin: 0 .75em 0 0;
}
p {
  display: inline-block;
  margin: 0;
}
</style>