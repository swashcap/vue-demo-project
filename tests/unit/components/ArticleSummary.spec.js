import {
  createLocalVue,
  shallowMount
} from '@vue/test-utils'
import VueRouter from 'vue-router'

import ArticleSummary from '../../../src/components/ArticleSummary.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

test('renders article summary', () => {
  const id = 101
  const summary = 'Burning up his fuse up here alone'
  const title = 'Rocket Man'
  const wrapper = shallowMount(ArticleSummary, {
    localVue,
    propsData: {
      id,
      summary,
      title
    },
    router
  })

  expect(wrapper.element).toMatchSnapshot()

  expect(wrapper.find('router-link-stub').attributes('to'))
    .toBe(`/article/${id}`)

  expect(wrapper.find('h1').text()).toBe(title)
  expect(wrapper.find('p').text()).toBe(summary)
})