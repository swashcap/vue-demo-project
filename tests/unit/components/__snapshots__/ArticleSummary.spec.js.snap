// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders article summary 1`] = `
<article>
  <router-link-stub
    event="click"
    tag="a"
    to="/article/101"
  >
    <h1>
      Rocket Man
    </h1>
     
    <p>
      Burning up his fuse up here alone
    </p>
  </router-link-stub>
</article>
`;
